--- Gtk2-Notify-0.05.orig/xs/Notify.xs	2007-10-04 15:11:13.000000000 +0300
+++ Gtk2-Notify-0.05/xs/Notify.xs	2011-12-14 16:02:19.218520417 +0200
@@ -85,6 +85,18 @@
 
 MODULE = Gtk2::Notify	PACKAGE = Gtk2::Notify	PREFIX = notify_notification_
 
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+
+NotifyNotification *
+notify_notification_new (class, summary, body=NULL, icon=NULL)
+		const gchar *summary
+		const gchar *body
+		const gchar *icon
+	C_ARGS:
+		summary, body, icon
+
+#else
+
 NotifyNotification *
 notify_notification_new (class, summary, body=NULL, icon=NULL, attach=NULL)
 		const gchar *summary
@@ -105,22 +117,6 @@
 	C_ARGS:
 		summary, body, icon, status_icon
 
-#endif
-
-gboolean
-notify_notification_update (notification, summary, message=NULL, icon=NULL)
-		NotifyNotification *notification
-		const gchar *summary
-		const gchar *message
-		const gchar *icon
-
-void
-notify_notification_attach_to_widget (notification, attach)
-		NotifyNotification *notification
-		GtkWidget *attach
-
-#if GTK_CHECK_VERSION (2, 9, 2)
-
 void
 notify_notification_attach_to_status_icon (notification, status_icon)
 		NotifyNotification *notification
@@ -129,12 +125,26 @@
 #endif
 
 void
+notify_notification_attach_to_widget (notification, attach)
+		NotifyNotification *notification
+		GtkWidget *attach
+
+void
 notify_notification_set_geometry_hints (notification, screen, x, y)
 		NotifyNotification *notification
 		GdkScreen *screen
 		gint x
 		gint y
 
+#endif
+
+gboolean
+notify_notification_update (notification, summary, message=NULL, icon=NULL)
+		NotifyNotification *notification
+		const gchar *summary
+		const gchar *message
+		const gchar *icon
+
 NO_OUTPUT gboolean
 notify_notification_show (notification)
 		NotifyNotification *notification
--- Gtk2-Notify-0.05.orig/t/notification.t	2006-08-27 22:11:57.000000000 +0300
+++ Gtk2-Notify-0.05/t/notification.t	2011-12-14 16:40:07.723402956 +0200
@@ -2,20 +2,18 @@
 
 use strict;
 use warnings;
-use Gtk2::TestHelper tests => 35;
+use Gtk2::TestHelper tests => 31;
 use Test::Exception;
 use Gtk2::Notify -init, $0;
 
 ginterfaces_ok('Gtk2::Notify');
 
-my $w = Gtk2::Window->new;
-my $n = Gtk2::Notify->new('foo', 'bar', '', $w);
+my $n = Gtk2::Notify->new('foo', 'bar', '');
 
 isa_ok($n, 'Gtk2::Notify');
 
 my @methods = qw(
         add_action
-        attach_to_widget
         clear_actions
         clear_hints
         close
@@ -46,16 +44,6 @@
         }, 42);
 }, 'add_action');
 
-{
-    my $nw = Gtk2::Window->new;
-    lives_ok(sub {
-            $n->attach_to_widget($nw);
-    }, 'attach_to_widget');
-    lives_ok(sub {
-            $n->attach_to_widget($w);
-    }, 'attach_to_widget');
-}
-
 lives_ok(sub {
         $n->clear_actions;
 }, 'clear_actions with existing actions');
@@ -96,12 +84,6 @@
 }, 'set_urgency');
 
 lives_ok(sub {
-        $n->close;
-}, 'close before show');
-
-$w->show_all;
-
-lives_ok(sub {
         $n->show;
 }, 'show');
 
