diff --git a/Makefile b/Makefile
index 28b0e99..446c37f 100644
--- a/Makefile
+++ b/Makefile
@@ -12,11 +12,14 @@ all: $(TARGETS)
 experimental: $(EXPERIMENTAL_TARGETS)
 
 install: all
+	$(INSTALL) -d $(PREFIX)/bin
 	$(INSTALL) -oroot -groot -m755 $(TARGETS) $(PREFIX)/bin
+	$(INSTALL) -d $(PREFIX)/share/man/man1
 	$(INSTALL) -oroot -groot -m644 f3read.1 $(PREFIX)/share/man/man1
 	$(LN) -sf f3read.1 $(PREFIX)/share/man/man1/f3write.1
 
 install-experimental: experimental
+	$(INSTALL) -d $(PREFIX)/bin
 	$(INSTALL) -oroot -groot -m755 $(EXPERIMENTAL_TARGETS) $(PREFIX)/bin
 
 f3write: utils.o f3write.o
